src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -480,6 +479,100 @@ public static unsafe bool TryParseInt64(StringSegment value, out long result)
             }
         }
 
+        // Fast RFC2616 3.9 Quality value parser (and without memory allocation)
+        internal static bool TryParseQualityDouble(string input, int startIndex, out double quality, out int length)
```

Comment:
We should optimize the opposite direction too, something like `string FormatQualityDouble(double quality)`